utils/types.ts

```diff
@@ -0,0 +1,14 @@
+import { LinkFragment } from "../generated/sdk";
+import { G } from "@mobily/ts-belt";
+
+type RequiredNotNull<T> = {
+  [P in keyof T]: NonNullable<T[P]>;
+};
+
+type SetNonNullable<T, K extends keyof T> = T & RequiredNotNull<Pick<T, K>>;
```

Comment:
golly, this was hard won. it was easy enough to fix specifically:

```ts
export type UsableLinkFragment = Omit<LinkFragment, "href"> & {
  href: NonNullable<LinkFragment["href"]>;
};
```

but, difficult to fix generically. I found some promising looking ideas (https://github.com/sindresorhus/type-fest/blob/main/source/set-required.d.ts#L29), but `NonNullable` always seemed to get lost when trying this approach. I still haven't solved that mystery to why, but I found salvation at stack overflow per usual: https://stackoverflow.com/a/61912075